ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/FSharpElementFactory.fs

```diff
@@ -210,3 +210,18 @@ type FSharpElementFactory(languageService: IFSharpLanguageService, psiModule: IP
             addParensIfNeeded rightArg |> ignore
 
             expr
+            
+        member x.CreateTypedPatInParens(typeSignature: string, parameterName: string) : IParenPat =
+            let expr = createLetBinding (sprintf "(%s : %s)" parameterName typeSignature)
+            let binding = expr.Bindings |> Seq.exactlyOne
+            binding.Children()
+            |> Seq.choose (function | :? IParenPat as pat -> Some pat | _ -> None)
```

Comment:
Please use `binding.HeadPattern`.

When in doubt about getters, I suggest navigating to a generated interface (like `IBinding` here) and it'll be easier to see what should be available. If there's something missing, it usually means it wasn't needed previously and can be added to the grammar that the generator use: https://github.com/JetBrains/fsharp-support/blob/net202/ReSharper.FSharp/src/FSharp.Psi/src/FSharp.psi#L178.